controller/characterController.go

```
@@ -0,0 +1,83 @@
+package controller
+
+import (
+	"os"
+	"encoding/csv"
+	"strconv"
+	"io"
+	
+	"github.com/carRub/academy-go-q12021/model"
+	"github.com/carRub/academy-go-q12021/usecase"
+)
+
+// CharacterController defines what a controller must contain
+type CharacterController struct {
+	characterInteractor usecase.CharacterInteractor
+}
+
+// CharacterUseCase defines functions to refer to our use cases
+type CharacterUseCase interface {
+	GetCharacters() ([]model.Character)
+	GetCharacterByID(id int) (*model.Character) 
+}
+
+// NewCharacterController Creates a new character controller 
+func NewCharacterController (ci usecase.CharacterInteractor) CharacterController {
+	return CharacterController{ci}
+}
+
+func (c CharacterController) GetCharacters() ([]model.Character, error) {
+	csvFile, _ := os.Open("assets/characters.csv")
+	defer csvFile.Close()
+	r := csv.NewReader(csvFile)
+
+	var character model.Character
+	var characters []model.Character
+
+	for {
+		record, err := r.Read()
+		
+		if err == io.EOF {
+			break
+		}
+
+		charID, _ := strconv.ParseInt(record[0], 10, 64)
```

As a recommendation, you should never ignore errors, and should handle them.

There are exceptions to the rule, and this is not a good example to do so. What if there's an error here? By default, it won't assign any value to `charID` if `.strconv.ParseInt` method fails, having the default value to 0, and be stored that way.

Is that what you want? Then it's okay, if not, please handle error according to your requirements.

---

This applies to the GetCharacterByID() method.